pkg/syncer/fullsync.go

```diff
@@ -31,12 +31,22 @@ import (
 	"sigs.k8s.io/vsphere-csi-driver/pkg/csi/service/logger"
 )
 
+var (
+	// migrationFeatureStateForFullSync is used to help determine workflow
+	// depending on migration feature is enabled or disabled
+	migrationFeatureStateForFullSync bool
```

Comment:
@chethanv28 instead of keeping global variable, can we supply feature status as an argument to following functions

getVolumeSpecs, getVolumesToBeDeleted, getEntityMetadata, getVolumesToBeDeleted

I have a concern about if someone starts using these functions in other workflows, reliance on this public variable will create ambiguity.

if these functions are only used by CSI full sync we can postfix them with `ForFullSync` for example `getVolumesToBeDeletedForFullSync`.

Reply:
+1

Reply:
I agree with supplying feature state as an argument.
However we need not rename the functions. These are private methods in fullsync. So, I think adding a suffix is not necessary